ballerina/tests/tests.bal

```diff
@@ -0,0 +1,313 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+import ballerina/lang.runtime;
+
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable string merchantEmail = ?;
+configurable boolean isLiveServer = ?;
+configurable string serviceUrl = isLiveServer ? "https://api-m.sandbox.paypal.com/v2/invoicing" : "http://localhost:9090";
+
+ConnectionConfig config = {
+    auth: {
+        clientId,
+        clientSecret
+    }
+};
+
+final Client paypalClient = check new (config, serviceUrl);
+
+string generatedInvoiceNumber = "";
+string testInvoiceId = "";
+string testPaymentId = "";
+string paymentTransactionId = "";
+
+@test:BeforeSuite
+function setup() returns error? {
+    InvoiceNumber result = check paypalClient->/generate\-next\-invoice\-number.post();
+    generatedInvoiceNumber = result.invoice_number ?: "";
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function CreateInvoice() returns error? {
+    test:assertNotEquals(generatedInvoiceNumber, "", msg = "Invoice number must be set before creating invoice");
+
+    Invoice invoicePayload = {
+        detail: {
+            invoice_number: generatedInvoiceNumber,
+            reference: "PO-123456",
+            invoice_date: "2025-06-17",
+            currency_code: "USD",
+           note: "Thanks for your business!",
+            memo: "Test invoice memo",
+            payment_term: {
+                term_type: "NET_30"
+            }
+        },
+        invoicer: {
+            name: {
+                given_name: "Dharshan",
+                surname: "Doe"
+            },
+            email_address: merchantEmail
+        },
+        primary_recipients: [
+            {
+                billing_info: {
+                    name: {
+                        given_name: "Ravindran",
+                        surname: "Smith"
+                    },
+                    email_address: "customer@example.com"
+                }
+            }
+        ],
+        items: [
+            {
+                name: "Item 1",
+                quantity: "2",
+                unit_amount: {
+                    currency_code: "USD",
+                    value: "25.00"
+                }
+            }
+        ]
+    };
+
+    Invoice result = check paypalClient->/invoices.post(invoicePayload);
+    test:assertNotEquals(result.id, "", msg = "Invoice ID should not be empty");
+    testInvoiceId = result.id ?: "";
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function ListInvoices() returns error? {
+    Invoices result = check paypalClient->/invoices.get(page = 1, page_size = 5, total_required = true);
+    int totalCount = 0;
+    if result.hasKey("total_items") {
+        anydata count = result["total_items"];
+        if count is int {
+            totalCount = count;
+        }
+    }

```

Comment:
Why the `total_count` is not the part of the type? Is it not present in the API spec?

Can the `total_count` field be `null`? If not, we can do the following.

```suggestion
    int totalCount = 0;
    int? totalItems = result.total_items
    if result.hasKey("total_items") {
        totalCount = check result["total_items"].ensureType();
    }

```